codec-http/src/test/java/io/netty/handler/codec/http/multipart/HttpPostRequestDecoderTest.java

```diff
@@ -976,4 +978,244 @@ public void testDecodeMultipartRequest() {
+            prefix[bp1.length] = brank[0];
+            prefix[bp1.length + 1] = brank[1];
+            for (int i = 0; i < bp2.length; i++) {
+                prefix[bp1.length + 2 + i] = bp2[i];
+            }
+            decoder.offer(new DefaultHttpContent(Unpooled.wrappedBuffer(prefix)));
+            byte[] body = new byte[bytesPerItem];
+            Arrays.fill(body, (byte) rank);
+            ByteBuf content = Unpooled.wrappedBuffer(body, 0, bytesPerItem);
+            decoder.offer(new DefaultHttpContent(content));
+            content.release();
+        }
+        byte[] lastbody = suffix.getBytes(CharsetUtil.UTF_8);
+        ByteBuf content2 = Unpooled.wrappedBuffer(lastbody, 0, lastbody.length);
+        decoder.offer(new DefaultHttpContent(content2));
+        content2.release();
+        decoder.offer(new DefaultLastHttpContent());
+        return decoder;
+    }
+
+    @Test
+    public void testNotBadReleaseBuffersDuringDecoding() throws IOException {
+        int nbItems = 20;
+        int bytesPerItem = 1000;
+        int maxMemory = 500;
+
+        String prefix1 = "\n--861fbeab-cd20-470c-9609-d40a0f704466\n" +
+                "Content-Disposition: form-data; name=\"image";
+        String prefix2 =
+                "\"; filename=\"guangzhou.jpeg\"\n" +
+                "Content-Type: image/jpeg\n" +
+                "Content-Length: " + bytesPerItem + "\n" + "\n";
+
+        String suffix = "\n--861fbeab-cd20-470c-9609-d40a0f704466--\n";
+
+        HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, "/upload");
+        request.headers().set("content-type", "multipart/form-data; boundary=861fbeab-cd20-470c-9609-d40a0f704466");
+        request.headers().set("content-length", nbItems * (prefix1.length() + prefix2.length() + 2 + bytesPerItem)
+                + suffix.length());
+
+        // First Test using Disk
+        HttpDataFactory factory = new DefaultHttpDataFactory(true);
+        HttpPostMultipartRequestDecoder decoder = commonNotBadReleaseBuffersDuringDecoding(factory, request, prefix1,
+                prefix2, suffix, nbItems, bytesPerItem, maxMemory);
+
+        for (int rank = 0; rank < nbItems; rank++) {
+            FileUpload data = (FileUpload) decoder.getBodyHttpData("image" + (10 + rank));
+            assertEquals(data.length(), bytesPerItem);
+            assertFalse(data.isInMemory());
+            byte[] body = new byte[bytesPerItem];
+            Arrays.fill(body, (byte) rank);
+            assertTrue(Arrays.equals(body, data.get()));
+        }
+        // To not be done since will load full file on memory: assertEquals(data.get().length, fileSize);
+        // Not mandatory since implicitely called during destroy of decoder
+        for (InterfaceHttpData httpData: decoder.getBodyHttpDatas()) {
+            httpData.release();
+            factory.removeHttpDataFromClean(request, httpData);
+        }
+        factory.cleanAllHttpData();
+        decoder.destroy();
```

Comment:
Same here with duplication and splitting into separate test cases.

Reply:
Again asserts are differents.
But agree to split into different tests using each a factory type.

Reply:
Done: split and refactoring tests